### ISSUE TITLE
Compilation errors do not change the exit status

### ISSUE THREAD

**Issue by samvasko** [oclint/oclint#439]
### Issue Summary
When an compilation error occurs the exit code remains zero.

### Environment
- Operation system name and version: 
```
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G1217
```
- OCLint version: 
```
LLVM (http://llvm.org/):
  LLVM version 4.0.0
  Optimized build.
  Default target: x86_64-apple-darwin15.6.0
  Host CPU: broadwell

OCLint (http://oclint.org/):
  OCLint version 0.12.
  Built Mar 27 2017 (20:27:21).
```
- How OCLint is installed: local build? prebuilt binary downloaded from github? homebrew install? others? homebrew

#### Sample code

```C
int main()
{
	return yes;
}
```
Run `oclint error-code.c`
  
### Expected Behavior
Failure should reflect in the exit status change
### Actual Behavior
exit status stays 0 indicating no error
